frontend/src/pages/VerificationPage.tsx

```diff
@@ -0,0 +1,72 @@
+import { useEffect, useRef } from "react";
+import { useLocation, useNavigate } from "react-router-dom";
+import { CircularProgress, Container } from "@mui/material";
+import { useSnackbar } from "notistack";
+
+import { useVerifyEmail } from "src/hooks/useAuth";
+import { ApiResponseError } from "src/services/ApiService";
+
+export const VerificationPage = () => {
+  const { search } = useLocation();
+  const params = new URLSearchParams(search);
+  const userId = params.get("userId");
+  const code = params.get("code");
+  const { enqueueSnackbar } = useSnackbar();
+  const navigate = useNavigate();
+
+  const previousCode = useRef<string>("");
+
+  const { verifyEmailMutation } = useVerifyEmail();
+
+  useEffect(() => {
+    if (userId === null || code === null) {
+      enqueueSnackbar("Something went wrong", {
+        variant: "error",
+        // Prevent double snackbar during development caused by
+        // React.StrictMode's double invocation
+        key: "Generic Error",
+        preventDuplicate: true,
+      });
+      navigate("/");
```

Comment:
Change navigation "/" to a button which resends email

Reply:
Will resolve in #107.